### PR TITLE
Add Coveralls step in GH actions

### PR DESCRIPTION
**Pull request by zeeke**
> Make code coverage profile output deterministic, in
> order to be collected by coveralls.io step.
> 
> Add coveralls badge to README.md
> 
> Signed-off-by: Andrea Panattoni <apanatto@redhat.com>

Coverage on this repository is [~77%](https://coveralls.io/github/zeeke/sriov-network-device-plugin). That's a very good percentage and this PR makes it recorded in future PRs.

If it is appreciated, I can add a coverage threshold to GH actions as described in [this blog post](https://medium.com/synechron/how-to-set-up-a-test-coverage-threshold-in-go-and-github-167f69b940dc).

### CODE REVIEW

**Review by e0ne — APPROVED** [review]
LGTM

**Comment by adrianchiris** [comment]
> @adrianchiris do we also want to add the ability for coverall to comment on the PR?

yes it would be best

File: .github/workflows/build-test-lint.yml [hunk 1]
```diff
@@ -40,7 +41,11 @@ jobs:
         run: sudo apt-get install hwdata -y
 
       - name: Go test
-        run: make test
+        run: make test-coverage
+      
+      - uses: shogo82148/actions-goveralls@v1
```
[the remark this answers is not in the document]
Comment (adrianchiris) [reply .github/workflows/build-test-lint.yml:46]: +1

[the remark this answers is not in the document]
Comment (zeeke) [reply .github/workflows/build-test-lint.yml]: I totally agree on using an official tool. I updated the action but I needed to add another third-party [tool for converting gcov to lcov](https://github.com/jandelgado/gcov2lcov).

**Comment by zeeke** [comment]
> https://docs.coveralls.io/coveralls-notifications

Just to be sure I didn't miss anything, notification instructions are all about configuring GH project. Nothing more to ad to the code. Right?

**Comment by SchSeba** [comment]
@zeeke is it possible to make the coverall a separate lane and leave the regular `make test` ?
this will be easier to check what unit test is failing

**Comment by zeeke** [comment]
Sure! good point

**Review by SchSeba — APPROVED** [review]

**Review by adrianchiris — APPROVED** [review]

**Comment by adrianchiris** [comment]
Thanks @zeeke !

**Comment by adrianchiris** [comment]
Folks, regarding configuration of GH project to allow coveralls to comment.

it seems i need to give it write access. are we OK with this ?

i didnt manage to get the alternative working

https://docs.coveralls.io/coveralls-notifications

**Pull request merged**